test/unit/overlay.js

```diff
@@ -0,0 +1,62 @@
+'use strict';
+
+var fixtures = require('../fixtures');
+var fs = require('fs');
+var sharp = require('../../index');
+
+sharp.cache(0);
+
+// Main
+describe('Overlays', function() {
+  it('Overlay transparent PNG on solid background', function(done) {
+    sharp(fixtures.inputPngOverlayLayer0)
+      .overlayWith(fixtures.inputPngOverlayLayer1)
+      .toBuffer(function (error, data, info) {
+        if (error) return done(error);
```

Comment:
(Just FYI, Mocha catches synchronous errors during async test execution. That's necessary since every assertion library, including Node's own, throws errors synchronously.)